### PR TITLE
Add micromamba as a possible install tool and make it default on macOS and Windows

### PR DESCRIPTION
**Pull request by jaimergp**
Checklist
* [x] Added a ``news`` entry
* [X] Regenerated schema JSON if schema altered (`python conda_smithy/schema.py`)

Investigating performance improvements triggered by conversation at https://github.com/conda-forge/conda-forge.github.io/issues/2323.

We can use `micromamba` to save some minutes, specially in Windows. To avoid installing Miniforge if possible, I scripted that too instead of hardcoding the logic in the CI workflow template. This opens the possibility of enabling `build-locally.py` on Windows, similar to macOS' approach.

Live tests available at https://github.com/conda-forge/dav1d-feedstock/pull/21

### CODE REVIEW

**Comment by beckermr** [comment]
pre-commit.ci autofix

**Comment by beckermr** [comment]
@conda-forge/core @jaimergp This one appears ready for review! Shall we merge?

**Review by beckermr — APPROVED** [review]

**Review by wolfv — APPROVED** [review]

**Comment by wolfv** [comment]
Awesome stuff, really looking forward to shave off these minutes from Windows! :)

**Comment by jaimergp** [comment]
We might need to change the `D:\` stuff on Windows so it's configurable (see #2076 for reference). I was hoping to get that one reviewed/merged, then rebase this one to use the things added there.

**Comment by beckermr** [comment]
Sounds good. We'll wait then.

**Comment by jaimergp** [comment]
This is getting quite fast now for Windows, under a minute! https://github.com/conda-forge/dav1d-feedstock/pull/21#issuecomment-2415359796

**Comment by beckermr** [comment]
Lgtm. Merge when you are ready.

**Comment by jaimergp** [comment]
The only rough edge here is that this doesn't affect Linux at all due to the Docker images. To benefit from micromamba there we would need to:

- Build and publish micromamba images
- Change the pinnings so the `docker_images` list refers to micromamba. Or maybe we post-process that by adding a clever mapping (e.g. prefixing everything with `micromamba-`).

Anyway, this is opt-in for now, so there's little risk.

**Comment by jaimergp** [comment]
Btw, same PR but for staged-recipes: https://github.com/conda-forge/staged-recipes/pull/27753

**Comment by beckermr** [comment]
Why not set this as the default?

**Comment by jaimergp** [comment]
To avoid the inconsistency between Linux and the rest, but in a way it's already inconsistent with Docker. We end up providing the same dependencies in the same base environment so I guess there's no point to be overly cautious. I'll flip the default and document this in the schema. Then folks can chime in if they have any thoughts.

File: conda_smithy/templates/run_osx_build.sh.tmpl [hunk 1]
```diff
@@ -40,11 +66,9 @@ export CONDA_SOLVER="{{ conda_solver }}"
 export CONDA_LIBMAMBA_SOLVER_NO_CHANNELS_FROM_INSTALLED=1
 {%- endif %}
 
+{% if conda_install_tool in ("conda", "mamba") -%}
```
[the remark this answers is not in the document]
Comment (jaimergp) [reply conda_smithy/templates/run_osx_build.sh.tmpl:69]: I'm anticipating that the next install tool is a self-contained environment provider like `pixi` and not another conda-like client that is first provided by a Miniforge-like installation, that's why I went with "special casing" these two.

**Comment by isuruf** [comment]
> Build and publish micromamba images

We can just put micromamba into the existing images. Won't make a difference

**Pull request merged**